app/src/main/java/example/chatea/servicecamera/PhonecallReceiver.java

```diff
@@ -0,0 +1,153 @@
+package example.chatea.servicecamera;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.ResultReceiver;
+import android.telephony.TelephonyManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.util.Date;
+
+public class PhonecallReceiver extends BroadcastReceiver {
+
+    //The receiver will be recreated whenever android feels like it.  We need a static variable to remember data between instantiations
+
+    private static int lastState = TelephonyManager.CALL_STATE_IDLE;
```

Comment:
I suggest to store the phone call state in Service. And this PhonecallReceiver only send the data into Service. The data in Service will not been killed, so you don't need to declared them as static value.